src/pages/profile.tsx

```diff
@@ -78,7 +108,88 @@ export default function Page() {
       <Input background="white" 
         value={newCity} onChange={e => setNewCity(e.target.value)} />
     </FormControl>           
+
+    <Divider margin={componentSpacing} />
+
+    <InterviewPreference data={unsaved?.interviews} 
+      updateData={updateInterviewPref} /> 
+
     <Button onClick={handleSubmit} variant="brand">保存</Button>
     {isLoading && <Loader loadingText='保存中...'/>}
   </VStack>;
 }
+
+function InterviewPreference({ data, updateData } : {
+  data: InterviewPref;
+  updateData: (data: InterviewPref) => void;
+}) {
+  const until = data?.limit?.until ? new Date(data.limit.until) : oneMonthDate;
+  const noMoreThan = data?.limit?.noMoreThan || 0;
+
+  const setLimit = (noMoreThan: number, until: Date) => {
+    if (noMoreThan !== undefined && until !== undefined) {
+      const newData = structuredClone(data || {});
+      newData.limit = { noMoreThan, until: until.toISOString() };
+      updateData(newData);
+    }
+  };
+
+  const removeLimit = () => {
+    if (data !== undefined) {
+      const newData = structuredClone(data);
+      delete newData.limit;
+      updateData(newData);
+    }
+  };
+
+  const toggleOptIn = (optIn: boolean) => {
+    updateData({
+      ...data,
+      optIn: optIn ? true : undefined,
+    });
+  };
+
+  return <>
+    <FormControl>
+      <FormLabel>面试官偏好</FormLabel>
+      <Checkbox isChecked={data?.optIn} 
+        onChange={e => toggleOptIn(e.target.checked)}>
+        我不是导师，但可以帮助面试学生。
+      </Checkbox>
+    </FormControl>
+
+    <FormControl>
+      <Flex wrap="wrap" alignItems="center">
+        <Checkbox isChecked={!!data?.limit}
+          onChange={e => e.target.checked ? setLimit(noMoreThan, until) : 
+            removeLimit()}>
+          面试限制：
+        </Checkbox>在
+        <DatePicker 
+          selected={until}
+          onChange={date => {
+            const newDate = date || oneMonthDate;
+            setLimit(noMoreThan, newDate);
+          }}
+          minDate={new Date()} maxDate={threeMonthsDate} disabled={!data?.limit}
+          customInput={<Input {...datePicker} textAlign="center" />}
+        />
+        之前，我还可以参与
+        <NumberInput 
+          value={noMoreThan} 
+          onChange={v => setLimit(Number.parseInt(v), until)}
+          min={0} max={10} isDisabled={!data?.limit} maxW="60px" marginX={1}
```

Comment:
specify margin in the theme file instead of here

Reply:
I put it into the theme file and don't know why it is not working. Maybe it is because of the maxW? Please can you check my codes.

Reply:
okay. in this case let's just remove non-working code in the theme file.